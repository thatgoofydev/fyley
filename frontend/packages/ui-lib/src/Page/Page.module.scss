@import "src/base-styling/colors";
@import "src/base-styling/vars";

$page-header-height: 70px;

.pageContainer {
  width: 100%;
  height: calc(100% - #{$page-header-height});
  max-width: 1200px;
  margin: 0 auto;

  .pageHeader {
    height: 70px;

    h1 {
      margin: 0;
      line-height: 70px;
      font-size: 24px;
      font-weight: bold;
    }
  }

  .pageContentWrapper {
    background-color: $white;
    height: 100%;
    min-height: 600px;
    -webkit-box-shadow: 0 0 20px 0 rgba(0,0,0,0.16);
    -moz-box-shadow: 0 0 20px 0 rgba(0,0,0,0.16);
    box-shadow: 0 0 20px 0 rgba(0,0,0,0.16);
  }
}