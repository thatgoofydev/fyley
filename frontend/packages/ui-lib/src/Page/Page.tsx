import React, { FunctionComponent } from "react";
import styles from "./Page.module.scss";

export interface IPageProps {
  title: string;
}

export const Page: FunctionComponent<IPageProps> = ({ title, children }) => {
  return (
    <div className={styles.pageContainer}>
      <header className={styles.pageHeader}>
        <h1>{title}</h1>
      </header>
      <div className={styles.pageContent}>{children}</div>
    </div>
  );
};
