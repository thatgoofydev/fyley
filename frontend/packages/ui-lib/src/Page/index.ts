export { Page } from "./Page";
export { PageContent } from "./PageContent";
export { PageError } from "./PageError";
