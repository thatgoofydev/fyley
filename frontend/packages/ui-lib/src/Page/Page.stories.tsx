import React from "react";
import { Meta, Story } from "@storybook/react";
import { IPageProps, Page } from "./Page";
import { PageContent } from "./PageContent";

export default {
  title: "Pages/Page",
  component: Page
} as Meta;

export const PageStory: Story<IPageProps> = args => {
  return (
    <Page {...args}>
      <PageContent>
        <p>Some content</p>
        <p>Some content</p>
      </PageContent>
    </Page>
  );
};
PageStory.storyName = "Page";
PageStory.args = {
  title: "Page title"
};
