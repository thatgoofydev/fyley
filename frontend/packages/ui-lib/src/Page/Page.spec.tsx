import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Page } from "./Page";

describe("Page", function () {
  it("should render", function () {
    const { container, getByText } = render(
      <Page title={"Page title"}>
        <p>Some content</p>
      </Page>
    );

    expect(getByText("Page title")).not.toBeNull();
    expect(getByText("Some content")).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
