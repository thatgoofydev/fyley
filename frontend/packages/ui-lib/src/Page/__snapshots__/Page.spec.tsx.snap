// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render 1`] = `
<div>
  <div
    class="pageContainer"
  >
    <header
      class="pageHeader"
    >
      <h1>
        Page title
      </h1>
    </header>
    <div
      class="pageContentWrapper"
    >
      <p>
        Some content
      </p>
    </div>
  </div>
</div>
`;
